pkg/server/server.go

```diff
@@ -153,8 +153,9 @@ type Server struct {
 	listObjectsDispatchThrottler throttler.Throttler
 	listUsersDispatchThrottler   throttler.Throttler
 
-	ctx                 context.Context
-	checkTrackerEnabled bool
+	ctx                         context.Context
+	checkTrackerEnabled         bool
+	datastoreContextPropagation bool
```

Comment:
Could you please also add 
```suggestion
	contextPropagationToDatastore bool
```